webclient/doi/model_serializers.py

```diff
@@ -0,0 +1,994 @@
+        publication_year_history = self._get_historie_queryset().filter(typ_zmeny=ARCHIVACE_SN)
+        if publication_year_history.exists():
+            return publication_year_history.first().datum_zmeny.year
+        else:
+            # DataCite request may be called before the history record is created
+            return datetime.now().year
+
+    def _get_title(self, language: str):
+        return {
+            "en": f"AMCR {self.record.ident_cely} – {self.record.druh_nalezu.heslo_en if self.record.druh_nalezu else ''} ({self.record.specifikace.heslo_en if self.record.specifikace else ''}), {self.record.obdobi.heslo_en if self.record.obdobi else ''}",
+            "cs": f"AMCR {self.record.ident_cely} – {self.record.druh_nalezu.heslo if self.record.druh_nalezu else ''} ({self.record.specifikace.heslo if self.record.specifikace else ''}), {self.record.obdobi.heslo if self.record.obdobi else ''}",
+        }[language]
+
+    def _serialize_alternate_identifiers(self):
+        alternate_identifiers = super()._serialize_alternate_identifiers()
+        if self.record.evidencni_cislo:
+            alternate_identifiers.append(
+                {
+                    "alternateIdentifierType": "Find inventory number",
+                    "alternateIdentifier": self.record.evidencni_cislo,
+                }
+            )
+        return alternate_identifiers
+
+    def _serialize_contributors(self):
+        return [serialize_osoba(self.record.projekt.vedouci_projektu, self.record.projekt.organizace, "ProjectLeader")]
+
+    def _serialize_creators(self):
+        return [serialize_osoba(author, self.record.projekt.organizace) for author in self._get_creators()]
+
+    def _serialize_dates(self):
+        dates = super()._serialize_dates()
+        dates.append({"date": self.format_date(self.record.datum_nalezu), "dateType": "Collected"})
+        return dates
+
+    def _serialize_descriptions(self):
+        descriptions = [{"lang": "en", "description": "Collected by field survey.", "descriptionType": "Methods"}]
+        if self.record.poznamka:
+            descriptions.append({"lang": "cs", "description": self.record.poznamka, "descriptionType": "Abstract"})
+        if self.record.okolnosti:
+            descriptions.append(
+                {
+                    "lang": "cs",
+                    "description": f"Finding context: {self.record.okolnosti.heslo_en}",
+                    "descriptionType": "Abstract",
+                }
+            )
+        if self.record.okolnosti:
+            descriptions.append(
+                {"lang": "cs", "description": f"Finding depth: {self.record.hloubka} cm", "descriptionType": "Abstract"}
+            )
+        if self.record.pocet:
+            descriptions.append(
+                {"lang": "cs", "description": f"Number of finds: {self.record.pocet}", "descriptionType": "Abstract"}
+            )
+        return descriptions
+
+    def _serialize_geo_locations(self):
```

Comment:
Není náhodou opomenuto ošetření situace, kdy přístupnost není anonym? Pokud ano, asi to bude platit i pro další situace (viz soubory s definicemi).

Reply:
Chybělo to, opraveno.